lib/pbench/server/api/resources/datasets_compare.py

```diff
@@ -21,14 +21,43 @@
     Schema,
 )
 from pbench.server.cache_manager import CacheManager
-from pbench.server.database.models.datasets import Metadata
+from pbench.server.database.models.datasets import Dataset, Metadata
 
 
 class DatasetsCompare(ApiBase):
     """
     This class implements the Server API used to retrieve comparison data for visualization.
     """
 
+    @staticmethod
+    def get_benchmark_name(dataset: Dataset) -> str:
+        """Convenience function to get dataset's benchmark
+
+        The Pbench Server intake constructs a server.benchmark metadata key to
+        represent the benchmark type. This helper implements a fallback for
+        datasets processed prior to the implementation of server.benchmark to
+        avoid null values, by using the Pbench Agent metadata.log "script"
+        value if that exists and then a constant fallback value.
+
+        TODO: This is a workaround from the transition to server.benchmark in
+        order to cleanly support earlier datasets on the staging server. This
+        can be removed at some point, but it's not critical.
```

Comment:
My instinct is that we cannot (or should not...) really rely on any particular item of metadata being present on the Server, unless the Server puts it there.  Given that we're expecting this value to be set by the client which we _expect_ to be the Pbench Agent (but it might not be!), I think the Server needs to be prepared for this value to be absent.  Thus, I think we'll always want at least the ability to fall back to `SERVER_BENCHMARK_UNKNOWN`, so I don't think there is any point to contemplating removing this function.

Reply:
That was kinda the point of `server.benchmark`. We were relying on the Agent-specific `metadata.log` `[pbench] script` value before, and now the server upload creates `server.benchmark` for each dataset: even if it's `server.archiveonly` or there's no `metadata.log`. It's always there, because the server creates it for that purpose.

Our problem today is that on the staging server we decided we still want to support datasets uploaded *yesterday* before we deployed that code. This provides a bridge; but after we no longer want to support those datasets (e.g., for a real 1.0 production server), this shim code will be irrelevant and wasteful.

Reply:
> It's always there, because the server creates it for that purpose.

Perfect.

Perhaps we should wipe the old datasets....